core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -421,6 +421,13 @@ object SparkHadoopUtil {
 
   val SPARK_YARN_CREDS_COUNTER_DELIM = "-"
 
+  // Just load HdfsConfiguration into the class loader to add
+  // hdfs-site.xml as a default configuration file otherwise
+  // some HDFS related configurations doesn't ship to Executors and
+  // it can cause UnknownHostException when NameNode HA is enabled.
+  // See SPARK-11227 for more details.
+  Utils.classForName("org.apache.hadoop.hdfs.HdfsConfiguration")
```

Comment:
Just reference it in any way, but, I guess we should ask, what does classloading do that we need, and is there any way to do that directly? this is fairly indirect. Is it that `    Configuration.addDefaultResource("hdfs-site.xml");` must be called?